src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -259,25 +286,29 @@ def render_definition_template(
         return ProjectProperties(None, {CONTEXT_KEY: {"env": environment_overrides}})
 
     project_context = {CONTEXT_KEY: definition}
+    template_env = TemplatedEnvironment(get_snowflake_cli_jinja_env())
 
     if "definition_version" not in definition or Version(
         definition["definition_version"]
     ) < Version("1.1"):
+        try:
+            referenced_vars = _get_referenced_vars_in_definition(
+                template_env, definition
+            )
+            if referenced_vars:
+                _template_version_warning(definition["definition_version"])
+        except Exception:
+            # also warn on Exception, as it means the user is incorrectly attempting to use templating
+            _template_version_warning(definition["definition_version"])
```

Comment:
Can you explain this? Generic exception -> version warning seems a bit odd at first glance. I'm worried we could end up masking something real in the future, for one.

Reply:
It's the same code that runs on both v1 and v1.1. If there is an issue hidden, it should appear with v1.1 templating.
We do not want to surface any accidental exceptions (if we miss to list an exception).

Reply:
Ok, so _get_referenced_vars_in_definition is 1.1+, so really if it fails you're assuming that the user was indeed using templates and should be warned about the PDF version? Is this because they could have an invalid template or something like that?

Reply:
yes exactly